### PR TITLE
feat(PF4 Dropdown): Wrap dropdown with context

### PR DESCRIPTION
**Pull request by karelhala**
**What**:
Wrap dropdown with context provider so we can change classes for each part of it. Consumers can now change basic classes if they want to. With this approach we will be able to reuse dropdown component as much as possible so we don't have to deal with different classes.

**Additional issues**:

### CODE REVIEW

**Comment by karelhala** [comment]
@jschuler do you know why circle CI fails from time to time?

**Comment by dgutride** [comment]
@karelhala - are you seeing an odd babelrc error that shows up in patternfly-3 packages even when they aren't changed?

**Comment by jschuler** [comment]
@karelhala not sure why it sometimes fails, I have not looked too much into that yet, @redallen would probably know! I restarted the circle ci workflow and has passed now

File: packages/patternfly-4/react-core/src/components/Dropdown/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export { default as Dropdown } from './Dropdown';
-export { DropdownDirection, DropdownPosition } from './dropdownConstants';
+export { default as Dropdown, DropdownWithContext as DefaultDropdown } from './Dropdown';
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/Dropdown/index.js]: Can you remove these exports from index.js.  We don't want these exposed to consumers.

**Pull request merged**